### PR TITLE
tests: make test_crash_on_closing_tmpfile_py27 more reliable

### PR DESCRIPTION
**Pull request by blueyed**
It fails reliable for me now without the fix from 9517c3a2a, and had
actually a wrong assertion in the end.

Ref: #2370

### CODE REVIEW

File: testing/test_capture.py [hunk 1]
```diff
@@ -1402,30 +1402,42 @@ def test_capattr():
...
 
         def test_silly():
+            global printing
+
             t = threading.Thread(target=spam)
             t.daemon = True
             t.start()
-            time.sleep(0.5)
 
+            while not printing.isSet():
```
Comment (nicoddemus) [inline testing/test_capture.py]: Perhaps `printing.wait()` instead?

Comment (nicoddemus) [reply testing/test_capture.py]: Hmm noticed this failed on Linux. You might try to only set the event after printing a number of times?

Comment (blueyed) [reply testing/test_capture.py]: For reference: all failed with `AssertionError: assert 'SPAMSPAMSPAM' in ''`.

Pushed a fixup based on your suggestions.

**Comment by blueyed** [comment]
It was not failing on CI anymore, but looked like it was not covered.  added a todo for diff coverage.

But maybe it would be better to just monkeypatch `f.close` to  raise IOError reliably?

**Comment by nicoddemus** [comment]
> But maybe it would be better to just monkeypatch f.close to raise IOError reliably?

Hmm that's an idea. Let's see, if this gets flaky we might go down that route, otherwise I'm fine with the code as it is.

**Comment by blueyed** [comment]
Ok, squashed.
Let's merge it when it stays green.

**Pull request merged**